src/BizHawk.Client.Common/EventTypes.cs

```diff
@@ -124,4 +150,18 @@ public StateSavedEventArgs(string stateName)
 	/// <param name="sender">Object that raised the event</param>
 	/// <param name="e">Event arguments</param>
 	public delegate void StateSavedEventHandler(object sender, StateSavedEventArgs e);
+
+	/// <summary>
+	/// Represent a method that will handle the event raised when a hard reset is performed
+	/// </summary>
+	/// <param name="sender">Object that raised the event</param>
+	/// <param name="e">Event arguments</param>
+	public delegate void HardResetEventHandler(object sender, HardResetEventArgs e);
+
+	/// <summary>
+	/// Represent a method that will handle the event raised when a soft reset is performed
+	/// </summary>
+	/// <param name="sender">Object that raised the event</param>
+	/// <param name="e">Event arguments</param>
+	public delegate void SoftResetEventHandler(object sender, SoftResetEventArgs e);
```

Comment:
Don't need custom delegate types, can use `EventHandler<>`, but also don't need custom arg struct with no props, so can use `Action`.

Reply:
Ok, I know @CasualPokePlayer below had suggested trying to poll  InputManager.ClickyVirtualPadController.IsPressed("Power") which doesn't work for me, but I'm seeing if he can give an example of something polling InputManager.ClickyVirtualPadController.IsPressed("Power")  or equivlent I can use for reference to fix my code instead.

 If his suggestion doesn't work, then I'll want to replace the delegates with a generic EventHandler<> class, correct? (Also, I'll remove the second e arg in all the calls in that event, I added them assuming that they should be in parity with existing callbacks, which was an incorrect assumption on my end -- I'll fix it if we need a follow up).

Reply:
`event Action HardReset;` in `IEmuClientApi` should suffice